extension/WebJobs.Extensions.RabbitMQ/Trigger/RabbitMQListener.cs

```diff
@@ -152,6 +158,30 @@ public ScaleStatus GetScaleStatus(ScaleStatusContext<RabbitMQTriggerMetrics> con
             return this.GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
         }
 
+        internal static Activity StartActivity(BasicDeliverEventArgs ea)
+        {
+            Activity activity;
+            if (ea.BasicProperties.Headers != null && ea.BasicProperties.Headers.ContainsKey("traceparent"))
+            {
+                byte[] traceParentIdInBytes = ea.BasicProperties.Headers["traceparent"] as byte[];
+                string traceparentId = Encoding.Default.GetString(traceParentIdInBytes);
```

Comment:
Why do we need to encode/decode the header value? I see RabbitMQ client encodes the string in UTF-8 by itself.

Reply:
When header value is string type, RabbitMQ convert it into byte array.